resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesSpec.scala

```diff
@@ -14,17 +14,18 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.apache.spark.deploy.k8s.submit.steps
+package org.apache.spark.deploy.k8s
 
-import org.apache.spark.deploy.k8s.submit.KubernetesDriverSpec
+import io.fabric8.kubernetes.api.model.HasMetadata
 
-/**
- * Represents a step in configuring the Spark driver pod.
- */
-private[spark] trait DriverConfigurationStep {
+private[spark] case class KubernetesSpec(
```

Comment:
This class is named as though it applies to driver and executor construction. Maybe `KubernetesDriverSpec`? It's also a bit unclear to me what purpose this abstraction serves as opposed to the way `KubernetesExecutorBuilder` goes about building the pod.

Reply:
If you check `KubernetesClientApplication`, it needs the extra fields here (the additional driver resources and the driver system properties) to construct the pod. In other words, the driver builder has to return a structure with more than just the pod.

Reply:
Think we could move some of the logic from `KubernetesClientApplication` into `KubernetesDriverBuilder`. Do you have any suggestions if we should move around the abstraction boundaries a bit?